src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -210,39 +231,43 @@ public override Task Save(DatabaseViewContext context, DatabaseInfo obj)
         public override Task<string> Script(DatabaseViewContext context, DatabaseInfo obj)
         {
             var script = ConfigureDatabase(
-                context.Parameters.ConnectionUri,
+                context.Parameters,
                 obj,
                 context.Parameters.IsNewObject ? ConfigAction.Create : ConfigAction.Update,
                 RunType.ScriptToWindow);
             return Task.FromResult(script);
         }
 
-        private CDataContainer CreateDatabaseDataContainer(string connectionUri, ConfigAction configAction, DatabaseInfo? database = null)
+        private CDataContainer CreateDatabaseDataContainer(InitializeViewRequestParams requestParams, ConfigAction configAction, DatabaseInfo? database = null)
         {
-            ConnectionInfo connectionInfo = this.GetConnectionInfo(connectionUri);
+            ConnectionInfo connectionInfo = this.GetConnectionInfo(requestParams.ConnectionUri);
             CDataContainer dataContainer = CDataContainer.CreateDataContainer(connectionInfo, databaseExists: configAction != ConfigAction.Create);
             if (dataContainer.Server == null)
             {
                 throw new InvalidOperationException(serverNotExistsError);
             }
-            string objectUrn = (configAction != ConfigAction.Create && database != null)
-                ? string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                    "Server/Database[@Name='{0}']",
-                    Urn.EscapeString(database.Name))
-                : string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                    "Server");
+            string objectUrn = requestParams.ObjectUrn;
+            if (!requestParams.IsNewObject)
+            {
+                objectUrn = ((configAction != ConfigAction.Create && database != null)
+                    ? string.Format(System.Globalization.CultureInfo.InvariantCulture,
+                        "Server/Database[@Name='{0}']",
+                        Urn.EscapeString(database.Name))
+                    : string.Format(System.Globalization.CultureInfo.InvariantCulture,
+                        "Server"));
```

Comment:
Still need to remove the extra outer layer of parentheses left over from the ternary operator.